### DISCUSSION TITLE
Can I register a conversion between SomeType<T> and T

### DISCUSSION THREAD

**Discussion by ronnyek** [LuckyPennySoftware/AutoMapper#3981] [NA]
We've got long standing code that effectively wraps a property with something like nullable, but not actually nullable. I'd like to take the mapping from one type that contains properties of say `SomeType<T>` and register a generic conversion/mapping that will automatically just grab `.Value` and return T so that theoretically

``` cs
public class BookInput{
   public SomeType<bool> IsFavorite{get;set;}
}
```
could be mapped to:
``` cs
public class Book{
   public bool IsFavorite{get;set;}
}
```

It looks like automapper supports mapping open generics, but tends to map from one Generic type to a different generic type, not a generic wrapper for the underlying type (unless I'm missing something).

I also realize I could just register each type like `SomeType<bool>` to `bool`, but I'm hoping to avoid all these manual registrations for every type, since the conversion really shouldnt be different.

**Comment by lbargaoanu** [comment]
If you have a conversion operator in your class, AM could use that.
